webhooks/validating/configuration/src/main.rs

```diff
@@ -789,6 +860,256 @@ mod tests {
         assert!(resp.allowed);
     }
 
+    #[test]
+    fn test_validate_configuration_discovery_properties_empty() {
+        let discovery_properties = "";
+
+        // no discovery properties specified should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_empty_list() {
+        let discovery_properties = r#"
+        "discoveryProperties": [],
+        "#;
+
+        // empty discovery properties array should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "value":"vvvv"
+            }
+        ],
+        "#;
+
+        // plain text discovery properties specified should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text_empty_value() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn",
+                "value": ""
+            }
+        ],"#;
+
+        // plain text discovery properties, empty value should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text_no_value() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": "nnnn"
+            }
+        ],"#;
+
+        // plain text discovery properties, value not specified should success
+        let resp = run_validate_configuration_discovery_properties(discovery_properties);
+        assert!(resp.allowed);
+    }
+
+    #[test]
+    fn test_validate_configuration_discovery_properties_plain_text_empty_name() {
+        let discovery_properties = r#"
+        "discoveryProperties": [
+            {
+                "name": ""
```

Comment:
curious, why this would be success case?

Reply:
we don't restrict empty name, as long as it's not None, we accept it (although it probably won't be used when DH query the secret data).

Reply:
seems strange to me to allow an empty name

Reply:
i think an empty name should be an error, but won't block the PR based on that.

Reply:
if we treat empty name as error, should we treat name with all spaces as error, too?  IMO, from syntax perspective, an empty string is a valid string and is valid as a key in key-value pair.  We can allow it.
BTW, a null string for name is for sure not allowed, it's specified in the schema

Reply:
i dont think an empty string and a string with spaces are the same.

Reply:
I'd be in favor of specifying name in the same way it is defined in core `EnvVar`, that is as a `C_IDENTIFIER` (i.e a string conforming to this regex: `[A-Za-z_][A-Za-z0-9_]*` thus disallowing empty names and ensuring these discovery properties can be passed as environment variable if need arise.

I don't know if there is a way to translate that in the CRD or if we can only rely on the webhook to reject configurations that are in this case.

Reply:
yes, we can put pattern in the crd and not relies on webhook to validate the content.  I'll investigate if we really want to make that hard restriction to the key name.

Reply:
filed Issue for the investigation effort.
https://github.com/project-akri/akri/issues/634